### ISSUE TITLE
Various review comments

### ISSUE THREAD

**Issue by falko-strenzke** [openpgp-pqc/draft-openpgp-pqc#6]
Regarding the sentence in the introduction:

> "Here, the term "composite" indicates that the combination of the two components forms a single atomic object.

I don't think that this sentence is saying something useful. Clearly, they are not essentially atomic (as nothing is in the known universe, not even atoms). What is meant here, is that they appear as single cryptogrpaphic schemes on the protocol layer. Accordingly, that is what I would propose to state here.

**Comment by falko-strenzke** [comment]
Section 4.1.1.1:

> Generate an ephemeral key pair {v, V=vG}, without any masking or clamping as defined in [[RFC7748](file:///home/fstrenzke/doc/pqc_thunderbird/draft-openpgp-pqc/draft-wussler-openpgp-pqc.html#RFC7748)]

I think this can be understood in two ways:
1. that the definition in RFC7748 should be used and that this definition does not include "masking or clamping"
2. that "masking or clamping" should not be used, and that "masking and clamping" is defined in RFC7748

In my understanding, the first variant is the intended statement. To make that entirely clear, I would rather write: "Generate an ephemeral key pair {v, V=vG} according to the procedure defined in RFC7748, that is, without any measure such as masking or clamping that are applied previous OpenPGP specifications of this scheme."

**Comment by falko-strenzke** [comment]
In Section 4.1.2 we state

> Extract the component public key kyberPublicKey that is part of the recipient's composite public key

This seems inconsistent in wording with the descriptions for the EC-KEMs, where no mentioning of the extraction of the component public key from the composite key is made. IMO it should be mentioned in for each scheme or for none.

**Comment by wussler** [comment]
> Section 4.1.1.1
> the first variant is the intended statement
Correct. The spec still defines clamping, but not in the key generation context.

> Generate an ephemeral key pair {v, V=vG} according to the procedure defined in RFC7748, that is, without any measure such as masking or clamping that are applied previous OpenPGP specifications of this scheme.

I like it! Mind making a PR?

**Comment by wussler** [comment]
> In Section 4.1.2
>This seems inconsistent in wording with the descriptions for the EC-KEMs, 

In the ECC sections we mention 
```...where R is the component public key eccPublicKey...```
I would consider removing point (1) from 4.1.2 and rewrite
``` 
2. Invoke (kyberCipherText, keyShare) <- kyberKem.encap(kyberPublicKey) 
   where kyberPublicKey is extracted from recipient's composite public key
```

**Comment by falko-strenzke** [comment]
In Section 4.2.3 we have a forward reference to Section 4.3.2. I think normally in a specification one would only make backwards references to things already defined. Thus the referenced section specifying the packet structure should better be moved in front IMO.

**Comment by falko-strenzke** [comment]
In Section 5.2.1 it says: 
> Given that Dilithium + ECC signature support is mandatory, an implementation MUST support SHA3-256 and SHOULD support SHA3-512.

The logic implied by "Given ..." is not entirely convincing to me, as it sounds as that the requirements stated the second clause sentence somehow naturally arise based on the first clause. But that seems not to hold, as an implementation could also just support one of them. So I would suggest to simply remove the first clause of that sentence, i.e. let the sentence start with "An implementations MUST ..."

**Comment by wussler** [comment]
> forward reference

I personally don't have anything against forward references. I usually like even the backrefs, so that you can navigate the spec in both directions.
Generally, when implementing, I don't use the spec linearly

**Comment by wussler** [comment]
> as an implementation could also just support one of them

Indeed. But here we define which one of the two is "more important".

We can rephrase it as
```
An implementation MUST support SHA3-256 and SHOULD support SHA3-512, since this specification mandates Dilithium + ECC signature support.
```

**Comment by falko-strenzke** [comment]
> > as an implementation could also just support one of them
> 
> Indeed. But here we define which one of the two is "more important".
> 
> We can rephrase it as
> 
> ```
> An implementation MUST support SHA3-256 and SHOULD support SHA3-512, since 
> this specification mandates Dilithium + ECC signature support.
> ```

I don't think that this rephrasing makes an essential difference. But it is also not a really severe point in my view, so maybe we simply leave it as it is.

**Comment by wussler** [comment]
I personally like it more, as it puts the focus on the MUST. We can also drop the "since..." but I would keep some justification for that must there.

What about
```
An implementation MUST support SHA3-256 and SHOULD support SHA3-512, in
order to support the hash binding with Dilithium + ECC signatures.
```

**Comment by falko-strenzke** [comment]
> I personally like it more, as it puts the focus on the MUST. We can also drop the "since..." but I would keep some justification for that must there.
> 
> What about
> 
> ```
> An implementation MUST support SHA3-256 and SHOULD support SHA3-512, in
> order to support the hash binding with Dilithium + ECC signatures.
> ```

Would also be OK for me.

**Comment by falko-strenzke** [comment]
Section 8.4:

> In order not to extend the attack surface, we bind the hash algorithm to the hash internally used into the signature algorithm.

I don't understand the meaning of the word "into" here. What is supposed to signify exactly? In my understanding, we are ensuring that the hash algorithm used for message digestion is the same as the one used internally in the signature algorithm.

**Comment by falko-strenzke** [comment]
In the Security Considerations, under Section 8.5, a normative statement is made ("MUST NOT"). In my understanding, normative statements do not belong in the Security Considerations, but under one of the normative sections.

**Comment by wussler** [comment]
This is technically already in the crypto-refresh: https://www.ietf.org/archive/id/draft-ietf-openpgp-crypto-refresh-07.html#section-12.2.3

I would like to reiterate here in the security considerations

**Comment by wussler** [comment]
> Section 8.4:
> 
> > In order not to extend the attack surface, we bind the hash algorithm to the hash internally used into the signature algorithm.
> 
> I don't understand the meaning of the word "into" here. What is supposed to signify exactly? In my understanding, we are ensuring that the hash algorithm used for message digestion is the same as the one used internally in the signature algorithm.

We can do `in` instead

**Comment by falko-strenzke** [comment]
In Section 5.2.2 and 5.2.3 there are statements like
> Sign dataDigest with eddsa.sign() from [Section 5.1.1]

I think it is somewhat unlucky to speak of "signing" here, because word this means "create a signature and add it to the signed data". However, how to add the signature to the packet is not specified here. This becomes clear only later when the signature packet structure is explained. 

I suggest to change the specification to saying "create an EdDSA signature with ..." and "create a Dilithium signature with ..." and then in a new step after these two: "encode the EdDSA and Dilithium signatures according to the the packet structure given in Section ...". But sure there also other possibilities to make the same statement.

**Comment by falko-strenzke** [comment]
> > Section 8.4:
> > > In order not to extend the attack surface, we bind the hash algorithm to the hash internally used into the signature algorithm.
> > 
> > 
> > I don't understand the meaning of the word "into" here. What is supposed to signify exactly? In my understanding, we are ensuring that the hash algorithm used for message digestion is the same as the one used internally in the signature algorithm.
> 
> We can do `in` instead

OK, and I would also suggest: "... we bind the hash algorithm used for message digestion to the hash algorithm used internally by signature algorithm."

**Comment by falko-strenzke** [comment]
I now made separate issues with all remaining open points.
